### PR TITLE
Distinguish api import functions from cython versions

### PR DESCRIPTION
**Pull request by da-woods**
Fixes issue with changed signature for these functions between Cython releases (https://github.com/cython/cython/pull/5289#issuecomment-1509371606)

### CODE REVIEW

File: Cython/Utility/ImportExport.c [hunk 1]
```diff
@@ -1,14 +1,3 @@
-/////////////// PyIdentifierFromString.proto ///////////////
```
Comment (da-woods) [inline Cython/Utility/ImportExport.c:1]: As far as I can tell, this is unused

File: Cython/Utility/ImportExport.c [hunk 2]
```diff
@@ -521,37 +510,39 @@ static int __Pyx_SetPackagePathFromImportLib(PyObject *module_name) {
 
 
 /////////////// TypeImport.proto ///////////////
+//@substitute: naming
 
-#ifndef __PYX_HAVE_RT_ImportType_proto
-#define __PYX_HAVE_RT_ImportType_proto
+#ifndef __PYX_HAVE_RT_ImportType_proto_$cy_version
```
Comment (dalcinl) [inline Cython/Utility/ImportExport.c:515]: Wouldn't it much nicer to use Tempita for this?

In the utility code you write
```
__PYX_HAVE_RT_ImportType_proto_{{HASH}}
```

and then you just load it with
```
TempitaUtilityCode.load_cached("TypeImport", "ImportExport.c", context={'HASH': cy_version})
```

Comment (dalcinl) [reply Cython/Utility/ImportExport.c:515]: Disregard my comment. Your approach works just fine, it is easier, and it is used everywhere.

**Comment by dalcinl** [comment]
@da-woods Would you mind adding the following nit? I find the output a bit hard to read because of these line breaks. The first hunk should not be objectionable. The second is a matter of taste, but one line per import with error checking at the end follows code generation patterns elsewhere. 

```patch
diff --git a/Cython/Compiler/ModuleNode.py b/Cython/Compiler/ModuleNode.py
index d64ff353d..491102c90 100644
--- a/Cython/Compiler/ModuleNode.py
+++ b/Cython/Compiler/ModuleNode.py
@@ -3850,7 +3850,7 @@ class ModuleNode(Nodes.Node, Nodes.BlockNode):
                 sizeof_objstruct, Naming.cy_version, sizeof_objstruct))
             code.putln("#endif")
         else:
-            code.putln('sizeof(%s), __PYX_GET_STRUCT_ALIGNMENT_%s(%s),' % (
+            code.put('sizeof(%s), __PYX_GET_STRUCT_ALIGNMENT_%s(%s),' % (
                 objstruct, Naming.cy_version, objstruct))
 
         # check_size
@@ -3861,7 +3861,7 @@ class ModuleNode(Nodes.Node, Nodes.BlockNode):
         else:
             raise RuntimeError("invalid value for check_size '%s' when compiling %s.%s" % (
                 type.check_size, module_name, type.name))
-        code.putln('__Pyx_ImportType_CheckSize_%s_%s);' % (
+        code.put('__Pyx_ImportType_CheckSize_%s_%s);' % (
             check_size.title(), Naming.cy_version))
 
         code.putln(' if (!%s) %s' % (type.typeptr_cname, error_code))
```

**Comment by scoder** [comment]
Could you rename the variable to cyversion? In the substitutions, it's unclear where the name ends, whether it's $cy or $cy_version.

**Review by dalcinl — APPROVED** [review]

File: Cython/Utility/ImportExport.c (hunk 1 above)
Comment (scoder) [reply Cython/Utility/ImportExport.c:1]: There used to be a `__Pyx_ImportModule()` that employed it. Probably still needed in 0.29.x.

**Review by scoder — APPROVED** [review]
Seems reasonable to do this. Avoids various possible issues regarding version incompatibilities. We should also backport this to 0.29.x, but keep the unused (in master) utility function.

File: Cython/Utility/ImportExport.c (hunk 1 above)
Comment (dalcinl) [reply Cython/Utility/ImportExport.c:1]: Looks like `__Pyx_PyIdentifier_FromString` is also unused in branch 0.29.x. 

```
$ git grep PyIdentifier
Cython/Compiler/ModuleNode.py:            h_code.put(UtilityCode.load_as_string("PyIdentifierFromString", "ImportExport.c")[0])
Cython/Utility/ImportExport.c://///////////// PyIdentifierFromString.proto ///////////////
Cython/Utility/ImportExport.c:#if !defined(__Pyx_PyIdentifier_FromString)
Cython/Utility/ImportExport.c:  #define __Pyx_PyIdentifier_FromString(s) PyString_FromString(s)
Cython/Utility/ImportExport.c:  #define __Pyx_PyIdentifier_FromString(s) PyUnicode_FromString(s)
```

**Pull request merged**